packages/app/pages/[modelName]/[documentTitle]/index.tsx

```diff
@@ -79,6 +87,53 @@ const EditDocumentPage = ({
         refreshDataInPlace(router)
     }, [formData.state, refreshDataInPlace])
 
+    useEffect(() => {
+        const privileges = user.privilegesForModelAndWorkflowNode(
+            modelName,
+            model.workflow.currentNode
+        )
+        // @ts-expect-error
+        const { userActivation } = globalThis.navigator
+
+        let intervalId: null | ReturnType<typeof globalThis.setInterval>
+        let consecutiveErrorCount = 0
+
+        if (userActivation) {
+            handleCollaborators()
+            intervalId = globalThis.setInterval(handleCollaborators, 3000)
+        }
+
+        async function handleCollaborators() {
+            // No need to keep triggering a re-render / sending a network request if collaboration isn't working.
+            if (consecutiveErrorCount > 5) {
```

Comment:
Since we're long polling, we don't want to keep calling the API if something is broken.